### DISCUSSION TITLE
그룹#3 회고

### DISCUSSION THREAD

**Discussion by godrm** [codesquad-members-2022/swift-pokergame#84] [Show and tell]
여기에 각자 이번주 회고를 기록해주세요.
- 학습하면서 잘했던 점, 좋았던 점은 무엇인가요? (다음주에도 계속 유지해보세요)
- 학습하면서 스스로 아쉬웠던 점, 부족했던 점은 무엇인가요 - 그래서 어떻게 개선할 수 있을까요? (다음주에는 개선해봐요)
- 새롭게 알게 되거나 시도하지 못했지만, 다음주에 도전해볼건 무엇인가요? (다음주에 시도해봐요)
- 학습하는 데 조금이라도 도움이 된 사람이 있다면 칭찬해주세요!

**Comment by Jeeehee** [comment]
* **잘했고 좋았던 점**
이번주도 열심히 학습일지를 작성했고, 질문을 많이했습니당

* **아쉽고 부족했던 점**
주 초반 열정 과다로, 주 후반에 사알짝 지친것같습니당. 집중이 잘 안되고 자꾸만 눕고싶어요,, 다음주엔 열정 배분 적절히,,,

* **다음주에 시도해볼것**
ARC를 학습만 하고 적용해보지 못했습니다.
또, 클래스와 구조체 사이에서 누굴 선택해야하는지 도무지 못정하겠읍니다,, 다음주엔 선택이 좀 더 명확해질 수 있게 학습을 보충하겠읍니다,,🥲

* **도움이 된 사람**
한 주 동기부여도 해주시고, 제 질문에 항상 친절히 알려주셨던 3조분덜 감사합니다,,👍
그리고 특히, Sol과 Jed 정말 감사드려요 !
질문공세에도 항상 미소를 잃지않고,, 제게 설명해주시곤 한번 더 확인해 알려주시는 미소천사 솔과
제가 질문했던 걸 잊지않고, 학습했던 것을 나눠주셨던! 이번주 저의 빅 웃음이었던 제드 감사합니다 🌟

**Reply by BumgeunSong** [reply]
![](https://t1.daumcdn.net/cfile/tistory/991188395CB055290C)

**Comment by BumgeunSong** [comment]
**학습하면서 잘했던 점, 좋았던 점은 무엇인가요? (다음주에도 계속 유지해보세요)**
- 월화수에 컨디션이 좋았고, 계획한 시간에 PR을 잘 보냈습니다.
- Gucci, Ebony, Jason, Chez와 만나서 밥먹고 같이 코딩한 게 기억에 남네요!
- 계속 써보려고 마음먹었던 Git 블로그 포스팅을 작성했습니다.
- 객체 설계하는 게 재밌습니다. 예전에 다각형 미션할 때 재미와 비슷한 느낌? 레고 블록 만드는 거 같고... 

**학습하면서 스스로 아쉬웠던 점, 부족했던 점은 무엇인가요 - 그래서 어떻게 개선할 수 있을까요? (다음주에는 개선해봐요)**

- 생활이 규칙적이었던 지난주와 다르게, 이번주는 약속이나 잠을 늦게 자는 등 생활 패턴이 많이 깨졌습니다. 집중하는 시간과 에너지에 안 좋은 영향을 많이 주더라고요ㅜ 다음주에는 꼭 일찍 자고 일찍 7-8시에 일어나는 사이클 가즈아ㅏ🔥
- 지난주만큼 다른 사람 코드 리뷰를 많이 읽어보지 못해서 아쉽네요.
- 줌 회의실에서 다른 분들을 많이 도와드리지 못했습니다. 다음주에는 회의실에서 열심히..!

**새롭게 알게 되거나 시도하지 못했지만, 다음주에 도전해볼건 무엇인가요? (다음주에 시도해봐요)**
- Test를 좀 더 잘하고 싶습니다. TEST부터 쓰려면 머리가 잘 안 돌아가는데, 주말-다음주에 TDD를 좀 더 시도해보고 손에 익혀야겠습니다.

**학습하는 데 조금이라도 도움이 된 사람이 있다면 칭찬해주세요!**
- [ 좋은 스터디](https://docs.google.com/spreadsheets/d/10tCBX_mltrUo37CeWD2GrezsMcTOStaPs4T65s-Qxgg/edit#gid=0)를 제안해준 Selina, Jed 감사 🙏

**Reply by sungju-kim** [reply]
에디 + TDD = ????

**Reply by BumgeunSong** [reply]
TEDDY로 진화

**Reply by BumgeunSong** [reply]
ㅈㅅ...

**Reply by junu0516** [reply]
깃 포스팅 기다립니다...!

**Comment by sungju-kim** [comment]
#### 학습하면서 잘했던 점, 좋았던 점은 무엇인가요? (다음주에도 계속 유지해보세요)
- 다음미션으로의 진행보다 리팩토링 위주의 학습을 했었는데, 처음엔 생각하지 못했던 방식에 대해서도 생각할수 있어서 좋았습니다.
####  학습하면서 스스로 아쉬웠던 점, 부족했던 점은 무엇인가요 - 그래서 어떻게 개선할 수 있을까요? (다음주에는 개선해봐요)
- 설계 → 구현 이 아니라 구현 → 리팩토링  이렇게 되니 수정을 많이하게되고, 수정을 하면 할수록 제가 이해할수 없는 코드가 되어간다고 느꼈습니다.
#### 새롭게 알게 되거나 시도하지 못했지만, 다음주에 도전해볼건 무엇인가요? (다음주에 시도해봐요)
- 전체적인 설계를 조금이라도 하고 구현을 해봐야겠습니다.
#### 학습하는 데 조금이라도 도움이 된 사람이 있다면 칭찬해주세요!
- 제 질문에 함께 고민해준 Sol, Jed 고마워요 👊👊

**Comment by AhnSangHee** [comment]
### 잘했던 점, 좋았던 점
- 수요일까지는 그래도 1일 1PR이라는 목표를 지켜냈다는 점
- 낮잠을 안 자기 시작했다는 점

### 아쉬웠던 점, 부족했던 점
- 어제 1일 1PR을 하지못했습니다.. 4단계부터 난항을 겪는중😭
- 목요일부터 지치는 바람에.. 체력관리랑 페이스 유지를 잘 해야할 것 같습니다

### 다음주에 도전해볼 것
- 규칙적인 생활하기
- 이번주 미션 다 끝내기,,

### 학습하는데 도움을 주신 분들
- 3조 모든 분들 감사합니다,,

**Comment by Hansolkkim** [comment]
[Sol]
## 학습하면서 잘했던 점, 좋았던 점은 무엇인가요? (다음주에도 계속 유지해보세요)
- 이번주부터 다 지키진 못 하더라도 계획을 일단 세우기 시작했고, 자기 전에 회고도 기록해보려고 노력했다.
- 기능을 나눈걸 말로만 듣다가, JK의 라이브 코딩을 보고 해당 개념이 조금이나마 명확해진 느낌을 받았다.
- 계획을 세워놓으니 PR을 보내고 merge 받기까지의 시간에 다른 할 일에 대해서도 좀 더 집중할 수 있었다.
## 학습하면서 스스로 아쉬웠던 점, 부족했던 점은 무엇인가요 - 그래서 어떻게 개선할 수 있을까요? (다음주에는 개선해봐요)
- 주말이 다가올수록 하루에 집중력이 떨어지는 시간이 빨라진다.. -> 억지로 앉아있끼 보다 쉴 땐 쉬면서 해도 괜찮다고 생각하기
- 코드 리뷰 내용 중에 잘 모르는 내용이 생기면, 구글링을 할 키워드 조차 떠올리지 못 할 때가 많았다,, -> 좀 더 적극적으로 질문해보기
- 코드로 앱 화면 구성하는걸 처음 해봤는데, 스토리보드로만 해봤더니 너무 어려웠다. 그럼 더 공부했어야 했는데 그러지 않았다.. -> 주말을 통해서라도 공부하고, 계속 회피하려고만 하지 않도록 노력해보자
## 새롭게 알게 되거나 시도하지 못했지만, 다음주에 도전해볼건 무엇인가요? (다음주에 시도해봐요)
- 이번에는 객체 내부의 프로퍼티에 제가 만든 객체를 사용하려고 할 때 `let` 상수로 선언하는게 더 좋다고 생각해서, 딜러, 참가자에 해당하는 객체들은 모두 class로 선언했습니다. 다음주 과제에서는 무조건 let 으로 선언하지 말고(무조건 class로 구현하지도 말고) 다른 방식으로 생각해보면서 구조를 짜봐야겠다고 생각했습니다.
## 학습하는 데 조금이라도 도움이 된 사람이 있다면 칭찬해주세요!
- 다 같이 힘내서 이번 과제 함께 해주신 3조 분들 다들 감사드립니다!!
- 아주 유익한 Swift 기술 면접 내용 스터디 제안해주신 Jed 감사해요
- 모든 분들의 코드 리뷰를 읽는건 힘들었는데, 코드 리뷰 내용 공유해주시면서 함께 고민하게 해주셨던 Dale 감사합니다!

**Reply by BumgeunSong** [reply]
> 스토리보드로만 해봤더니 너무 어려웠다. 그럼 더 공부했어야 했는데 그러지 않았다.. 

자기를 더욱더 채찍질하는 열쩡남... 천천히 해유

**Reply by Hansolkkim** [reply]
> 자기를 더욱더 채찍질하는 열쩡남... 천천히 해유

ㅋㅋㅋㅋㅋ생각만하고 안하죠 당연히

**Comment by junu0516** [comment]
### 좋았던 점과 아쉬운 점
- 좋았던 점 : 
    - 지난주보다 코드 작성 및 로직 설계 자체에 대한 피드백을 더 많이 받아서 생각하지 못했던 부분에 대해 많이 학습해볼 수 있었다. 
    - 특히 별도의 타입을 선언해서 배열이나 다른 원시타입 프로퍼티를 한 번 감싸주는 것에 대해 많은 고민을 하였다.
- 아쉬웠던 점 : 
    - 커밋 메시지를 일관성없게 작성하다보니 스스로도 읽기가 힘들다는 생각이 많이 들었다. 다음주에는 좀 더 읽기 편하게 일관성 있는 기준을 세워서 커밋 단위를 나누고 메시지를 정확히 작성해야 겠다.
    - 좀 늦게 자서 오전에 비몽사몽한 상태로 시간을 버린 날이 있었는데 다음 주에는 미라클 모닝을 실천해서 오전을 좀 더 생산적으로 보낼 수 있도록 해야겠다.
    
### 다음주에 도전해볼 것
- 일찍 자고 일찍 일어나기
- 좀 더 많이 고민해보고 Pull Request 날릴 때 많은 질문 해보기
    
### 학습하는 데 도움을 주신 분들
- 객체지향 설계하면서 좀 사소하거나 시덥지 않은 부분까지 물어봤는데, 같이 고민해주고 많은 해결책을 제시해주신 팀원들에게 모두 감사합니다!
- 수업과 별도로 swift 언어 자체에 대해 좀 더 공부해보고 싶었는데 스터디를 선뜻 같이 해주신 것도 감사합니다~~!

**Reply by BumgeunSong** [reply]
제드 짱
![](http://cdnimage.dailian.co.kr/news/201106/news1307017947_249767_1_m.jpg)

**Reply by junu0516** [reply]
마이크 항상 조심!

**Comment by wnsxor1993** [comment]
- 잘했고 좋았던 점
스트레스를 풀기 어려운 상황에서도 나름 잘 버텼다는 게 어느정도 잘했다고 볼 수 있지 않을까 생각합니다

- 아쉽고 부족했던 점
한 주동안 컨디션을 잘 배분해서 공부를 했어야 했는데 너무 초반에 쏟아넣었는지 후반부에 들어서 의욕이 많이 떨어져버렸습니다ㅠㅠ

- 다음주에 시도해볼것
컨디션의 적절한 배분!

- 도움이 된 사람
여러모로 힘든 한 주였지만 3조 분들과 함께여서 다행히 버틸 수 있었습니다ㅎㅎ
주저리주저리 질문이나 한풀이도 잘 들어주신 3조 분들 짱짱맨!!ㅎㅎ
